### DISCUSSION TITLE
fcst Lead Times in Stat Analysis Wrapper

### DISCUSSION THREAD

**Discussion by CPKalb** [dtcenter/METplus#1639] [Configuration] [answered]
Hi:

I'm trying to run stat_analysis, using the wrapper, retaining only certain lead times.  I have set in my .conf file:

FCST_INIT_HOUR_LIST = 00, 06, 12, 18
FCST_LEAD_LIST = 01, 02, 03, 04, 05, 06, 07, 08, 09, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24
GROUP_LIST_ITEMS = FCST_LEAD_LIST
LOOP_LIST_ITEMS = FCST_INIT_HOUR_LIST

The log file suggests it's retaining all 48 hours.
DEBUG 2: Computing output for 250 case(s).
DEBUG 2: For case "11:PMSL", found 48 unique FCST_LEAD values: 010000,020000,030000,040000,050000,060000,070000,080000,090000,100000,110000,120000,130000,140000,150000,160000,170000,180000,190000,200000,210000,220000,230000,240000,250000,260000,270000,280000,290000,300000,310000,320000,330000,340000,350000,360000,370000,380000,390000,400000,410000,420000,430000,440000,450000,460000,470000,480000

However, the output from stat analysis reports a lead of 480000.  There are lead times out to 48 hours in the lookin directory, but I'm only wanting to accumulate over the 1 - 24 hour lead times.  Is there another way to get stat_analysis to filter out these longer lead times?

Thanks,
Tina

**Comment by georgemccabe** [comment]
@malloryprow any advice here? We may need to see the full log file to figure out what exactly is happening and why it is not filtering the forecast leads properly.

**Reply by CPKalb** [reply]
Thanks.  I'm attaching a log file.  I cut it down to 1 run of stat_analysis to make things easier.  The lead times show up in the log file, but I'm not getting any filtering.  So, I don't know what I'm doing incorrectly.
[metplus.log.20220602193340.txt](https://github.com/dtcenter/METplus/files/8827358/metplus.log.20220602193340.txt)

**Comment by CPKalb** [comment, marked as answer]
I've just tried adding the following to my .conf file:
STAT_ANALYSIS_CONFIG_FILE = {CONFIG_DIR}/STATAnalysisConfig_wrapped

Now, I get a different error:
ERROR  :
ERROR  : yyerror() -> syntax error in file "/lus/work/wrf/kalb/met/tmp/met_config_118109_0"
ERROR  :
ERROR  :    line   = 15
ERROR  :
ERROR  :    column = 9
ERROR  :
ERROR  :    text   = ","
ERROR  :
ERROR  :
ERROR  : fcst_lead = "010000", "020000", "030000", "040000", "050000", "060000", "070000", "080000", "090000", "100000", "110000", "120000", "130000", "140000", "150000", "160000", "170000", "180000", "190000", "200000", "210000", "220000", "230000", "240000";
ERROR  : ________^__________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________
ERROR  :

I'm guessing I have to specify the stat_analysis config file for the forecast lead times to work.  Do we possibly need brackets around the lead list?

**Reply by georgemccabe** [reply]
I think you have figured it out. StatAnalysis wrapper is one of the few wrappers that allows you to run it with or without a config file. When no config file is specified, the METPLUS_FCST_LEAD environment variable is essentially ignored. You are correct that brackets are needed. There is a bug in the wrapper. I'm guessing the wrapper had only be run previously using a single forecast lead for each run, so this was never encountered.

Could you create a GitHub issue for this fix and post the link here?

**Reply by georgemccabe** [reply]
Also, if there is a branch you are currently working in for this use case, let me know the name. I can apply this fix to that branch so you can use this use case to test the changes.

**Reply by CPKalb** [reply]
Yes I will create an issue.  I don't have a branch I'm working out of for this yet, but I will need to make one.  Should I make one or will it be better to have one connected to the issue?

**Reply by georgemccabe** [reply]
I would create the issue, then I can create a branch for the fix using the issue number. Then you can get that branch to test the changes. You can create a branch off of that one if you need for the eventual PR to add your new use case.

**Reply by CPKalb** [reply]
I created the issue.  I can provide some sample data if that is helpful.  Just let me know where to put that data.  Who should I assign the issue to?

**Reply by georgemccabe** [reply]
You can assign it to both of us. I can make the change, then you could test that it works with your use case.

**Comment by CPKalb** [comment]
Here is a link to the issue to fix the missing brackets around the lead list:

https://github.com/dtcenter/METplus/issues/1641
